app/models/lookup_keys/lookup_key.rb

```diff
@@ -14,65 +14,39 @@ class LookupKey < ActiveRecord::Base
 
   serialize :default_value
 
-  belongs_to :puppetclass, :inverse_of => :lookup_keys, :counter_cache => true
-  has_many :environment_classes, :dependent => :destroy
-  has_many :environments, :through => :environment_classes, :uniq => true
-  has_many :param_classes, :through => :environment_classes, :source => :puppetclass
-  def param_class
-    param_classes.first
-  end
-
-  def audit_class
-    param_class || puppetclass
-  end
-
   has_many :lookup_values, :dependent => :destroy, :inverse_of => :lookup_key
   accepts_nested_attributes_for :lookup_values,
                                 :reject_if => :reject_invalid_lookup_values,
                                 :allow_destroy => true
 
-  before_validation :cast_default_value, :unless => Proc.new{|p| p.use_puppet_default }
-  validates :key, :uniqueness => {:scope => :is_param }, :unless => Proc.new{|p| p.is_param?}
+  before_validation :cast_default_value
 
   validates :key, :presence => true
-  validates :puppetclass, :presence => true, :unless => Proc.new {|k| k.is_param?}
   validates :validator_type, :inclusion => { :in => VALIDATOR_TYPES, :message => N_("invalid")}, :allow_blank => true, :allow_nil => true
   validates :key_type, :inclusion => {:in => KEY_TYPES, :message => N_("invalid")}, :allow_blank => true, :allow_nil => true
   validate :validate_default_value
   validates_associated :lookup_values
-  validate :ensure_type, :disable_merge_overrides, :disable_avoid_duplicates, :disable_merge_default
+  validate :disable_merge_overrides, :disable_avoid_duplicates, :disable_merge_default
 
   before_save :sanitize_path
   attr_name :key
 
-  scoped_search :on => :key, :complete_value => true, :default_order => true
-  scoped_search :on => :lookup_values_count, :rename => :values_count
-  scoped_search :on => :override, :complete_value => {:true => true, :false => false}
-  scoped_search :on => :merge_overrides, :complete_value => {:true => true, :false => false}
-  scoped_search :on => :merge_default, :complete_value => {:true => true, :false => false}
-  scoped_search :on => :avoid_duplicates, :complete_value => {:true => true, :false => false}
-  scoped_search :in => :param_classes, :on => :name, :rename => :puppetclass, :complete_value => true
-  scoped_search :in => :lookup_values, :on => :value, :rename => :value, :complete_value => true
+  def self.inherited(child)
```

Comment:
tbh: I'm not sure why this is needed to be eval on the child, if its a common attribute its fine to leave it in the base class, then specific cases can go to the sub class?